### ISSUE TITLE
Request: Option(s) to show all header lines in article view unwrapped

### ISSUE THREAD

**Issue by mglh** [newsboat/newsboat#687]
Since some articles have header lines (title, URL, ...) with one line and others with more, the place of the first line of the article description jumps when reading through the articles. For me, that disturbs the reading flow.

Therefore, it would be nice to have an option which allows the header lines to be shown unwrapped (and truncated) in article view. Possibly for each header line seperately.

For URLs this may have security implications, as later parts of an URL would be invisible, but for me that would be reasonable.

**Comment by Minoru** [comment]
Hi!

All the entries in the header—feed title, article title, article author etc.—are optional, so lengthy title isn't the only thing that can affect header's "height". Thus, not wrapping the header will only fix part of the problem.

Also, while I can see how this is a problem for the header, I don't understand why should links at the bottom not be wrapped. Right now they kinda aren't—`text-width` doesn't affect them, they're always wrapped at the screen's border. Can you explain?

The only solution I can think of is to fix the header's "height", but that's lame. Any other ideas?

(Tagging as "question", not "enhancement", because there's no clear path forward on this.)

**Comment by mglh** [comment]
> All the entries in the header -- feed title, article title, article author etc. --
> are optional, 

I see, didn't know that.

> so lengthy title isn't the only thing that can affect header's "height". 
> Thus, not wrapping the header will only fix part of the problem.

At least, can one assume that the existence of the header items is consistent over all articles in its respective feed? If the header of all articles in a certain feed would take up the same number of lines, by not wrapping long ones, "jumps" in the sense I described above would only occur when switching from one feed to an other. And even that could be avoided by replacing empty values with a string like "\<undefined\>", if the user desires it. 

> Also, while I can see how this is a problem for the header, 
> I don't understand why should links at the bottom not be wrapped. [...]
> Can you explain?

You are right, there is no need to not wrap (and truncate) the links at the bottom. Nonetheless, I would vote for not wrapping them for reasons of "optical consistency" between them and the article header. This could be made choosable by the user, too.

Does that answer your question?

I know, I am asking for a lot =:)

**Comment by Minoru** [comment]
> At least, can one assume that the existence of the header items is consistent over all articles in its respective feed?

No, because some items depend entirely on the article—its title, its author, its flags, it podcast URL and type; maybe even more, but I didn't check the code.

> And even that could be avoided by replacing empty header values with a string like "\<undefined>", if the user desires it.

I still find this lame. Firstly, this is a round-about way of solving the problem ("I want the article body to start at the same line every time, so let me pad the header to have a fixed height"). Secondly, I just don't think it's a good idea to make our internal renderer very complicated. I think it should serve the needs of the most, and the rest can use some extension mechanism (like `html-renderer`) to tweak things to their exact specification.

But article headers don't have anything like `html-renderer` at the moment.

If I take this idea to the extreme, and mix in #688 for a good measure, it appears that Newsboat needs a templating engine that lets the user decide how to lay out the article. Right now this seems like an overkill to me, but it'll probably solve this whole class of problems.

You did answer my question; thanks. I'm also sorry, because the "optical consistency" argument is dead to me. Newsboat isn't meant to be beholden from the distance—the user is supposed to *read* what Newsboat displays. So truncating lines just to look pretty gets outright "no".

**Comment by mglh** [comment]
Thank you for your answer. Your code, your rules =;)

Still, if you ever would think about giving the header(s) a fixed height, it would be appreciated. I get about 400 new articles each day, and with a "motionless" layout the reading experience would be even more enjoyable, at least for me.

You may close this issue now.

**Comment by Minoru** [comment]
Sorry I'm not ready to do anything about this one yet. I'll keep it open, since the problem wasn't solved and I don't consider this a "wontfix".

I won't close #688 either, because I think there is a solution that would satisfy both of us. It's a matter of implementing it now.

Thank you for taking the time to file both of those.
